dev-tools/omdb/src/bin/omdb/db.rs

```diff
@@ -3672,6 +3700,140 @@ async fn cmd_db_region_find_deleted(
     Ok(())
 }
 
+async fn cmd_db_dry_run_region_allocation(
+    opctx: &OpContext,
+    datastore: &DataStore,
+    args: &DryRunRegionAllocationArgs,
+) -> Result<(), anyhow::Error> {
+    let volume_id = VolumeUuid::new_v4();
+
+    let size: external::ByteCount = args.size.try_into()?;
+    let block_size: params::BlockSize = args.block_size.try_into()?;
+
+    let (blocks_per_extent, extent_count) = DataStore::get_crucible_allocation(
+        &block_size.try_into().unwrap(),
+        size,
+    );
+
+    let allocation_strategy = if args.distinct_sleds {
+        RegionAllocationStrategy::RandomWithDistinctSleds { seed: None }
+    } else {
+        RegionAllocationStrategy::Random { seed: None }
+    };
+
+    let query = match nexus_db_queries::db::queries::region_allocation::allocation_query(
+        volume_id,
+        args.snapshot_id,
+        nexus_db_queries::db::queries::region_allocation::RegionParameters {
+            block_size: args.block_size.into(),
+            blocks_per_extent,
+            extent_count,
+            read_only: args.snapshot_id.is_some(),
+        },
+        &allocation_strategy,
+        args.num_regions_required,
+    ) {
+        Ok(q) => q,
+        Err(e) => {
+            eprintln!("error from allocation_query: {e:?}");
+            bail!("error from allocation_query");
+        }
+    };
+
+    let err = OptionalError::new();
```

Comment:
Might be worth a comment, we're basically abusing this type to return a non-error value, right?

(I think that's fine, but it's counterintuitive that the thing being passed here is the **real result** of the query, not a typical error type)

Reply:
We are yeah - 33829c391 changes this to explicitly reference a "Result" enum